rct229/rulesets/ashrae9012019/section4/section4rule2.py

```diff
@@ -278,8 +263,10 @@ def get_fail_msg(self, context, calc_vals=None, data=None):
                 maximum_humidity_stpt_sch_id_b = calc_vals[
                     "maximum_humidity_stpt_sch_id_b"
                 ]
+
+                # guaranteed that either b or p has Null
                 return (
-                    FAIL_MSG_B
-                    if minimum_humidity_stpt_sch_id_b or maximum_humidity_stpt_sch_id_b
-                    else FAIL_MSG_P
+                    FAIL_MSG_P
+                    if (minimum_humidity_stpt_sch_id_b and maximum_humidity_stpt_sch_id_b)
```

Comment:
I think this should be or. Suppose `minimum_humidity_stpt_sch_id_b` exists and `minimum_humidity_stpt_sch_id_p` does not exist. Meanwhile, `minimum_humidity_stpt_sch`does not exist in both baseline and proposed. This is a failure case where `FAIL_MSG_B = (
    "Fail because a humidity schedule is defined in the B_RMR but not in the P_RMR."
)` applies. Could you double-check? @weilixu Thanks

Reply:
The naming of FAIL_MSG_B and FAIL_MSG_P maybe misleading, we may switch them

Reply:
I think the parameter name is confusing. I got this swapped but I think It is `and`.
Note this is failed message - preconditioned needs to take into consideration from `manual_check_required` and `rule_check`.
At this step, we already know schedule id data type are mismatched because:
1. manual check returns True if schedule data type matched and schedule value mismatched
2. rule check returns True if schedule data type matched and schedule value matched.

The only failure that comes this far is schedule data type mismatched. Basically, it means the `humidity_schedule_id_b` and `humidity_schedule_id_p` have different data type (one is None, one has value). 
Note - if a schedule id is not exist, then the schedule won't exist, this is another pre-condition checked by non-schema validation.

So in that case, what I need to know at this point is simple which model does not have the schedule. Is it baseline or proposed?
So in this case, I will say either `minimum_humidity_stpt_sch_id_b` or `maximum_humidity_stpt_sch_id_b` is None (`(minimum_humidity_stpt_sch_id_b is None) or (maximum_humidity_stpt_sch_id_b is None)` = `minimum_humidity_stpt_sch_id_b  and maximum_humidity_stpt_sch_id_b`), then we return the `The schedule is defined in proposed, but not in baseline.`

Reply:
Oh, I don't think the message is swapped.
`if minimum_humidity_stpt_sch_id_b  and maximum_humidity_stpt_sch_id_b` means proposed is missing but baseline has. So the message is correct `FAIL_MSG_B`

Reply:
`So in this case, I will say either minimum_humidity_stpt_sch_id_b or maximum_humidity_stpt_sch_id_b is None ((minimum_humidity_stpt_sch_id_b is None) or (maximum_humidity_stpt_sch_id_b is None) = minimum_humidity_stpt_sch_id_b  and maximum_humidity_stpt_sch_id_b), then we return the The schedule is defined in proposed, but not in baseline.`, everything makes sense before this. I tested locally and if we use `and`, case 4 and 5 would return the same msg. It took me quite a while to realize that there could be some complex case, such as `minimum_humidity_stpt_sch` exists in baseline but not in proposed, plus `maximum_humidity_stpt_sch` exists in proposed but not in baseline, then the msg can be `FAIL_MSG_B` or `FAIL_MSG_P`

Reply:
I see. The confusing part is RDS is trying to condense four logic branches (maximum/minimum and baseline/proposed) into two logic branches. Reading RDS languages again, I think it uses `or` which means as long as there is one type of schedule existing in baseline, then returning `FAIL_MSG_G`.
The `and` takes another approach (both types of schedules exist in baseline to return `FAIL_MSG_G`), which is indeed different from the RDS.

I will update the rules to reflect the RDS. thanks.